MODINVSTOR-343: Add preceding-succeeding relationship. Add sample data.

bohdan-suprun opened the pull request:
* Add preceding-succeeding relationship type
* Add tests 
* Add sample data

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: mod-inventory-storage/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 72 JSON schema files under directory './ramls'
INFO: lint-raml-cop: servicepoint.json: Each property "description" is present.
ERROR: lint-raml-cop: uuid.json: Missing top-level "description".
INFO: lint-raml-cop: instancerelationships.json: Each property "description" is present.
INFO: lint-raml-cop: materialtype.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationship.json: Each property "description" is present.
INFO: lint-raml-cop: instanceformat.json: Each property "description" is present.
INFO: lint-raml-cop: materialtypes.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterms.json: Each property "description" is present.
INFO: lint-raml-cop: classificationtype.json: Each property "description" is present.
INFO: lint-raml-cop: instancetype.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: loclib.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsrecords.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetype.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatus.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertypes.json: Each property "description" is present.
INFO: lint-raml-cop: contributortypes.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetypes.json: Each property "description" is present.
ERROR: lint-raml-cop: item.json: Missing "description" for: barcode, chronology, copyNumbers, descriptionOfPieces, electronicAccess, enumeration, holdingsRecord2, holdingsRecordId, itemDamagedStatusDate, itemDamagedStatusId, itemIdentifier, itemLevelCallNumber, itemLevelCallNumberPrefix, itemLevelCallNumberSuffix, itemLevelCallNumberTypeId, materialTypeId, missingPieces, missingPiecesDate, numberOfMissingPieces, numberOfPieces, permanentLoanTypeId, permanentLocationId, staffOnly, temporaryLoanTypeId, temporaryLocationId, volume, yearCaption
INFO: lint-raml-cop: instanceformats.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertype.json: Each property "description" is present.
INFO: lint-raml-cop: instancetypes.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatuses.json: Each property "description" is present.
INFO: lint-raml-cop: servicepointsusers.json: Each property "description" is present.
INFO: lint-raml-cop: instances.json: Each property "description" is present.
INFO: lint-raml-cop: loclibs.json: Each property "description" is present.
INFO: lint-raml-cop: loccamps.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationship.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterm.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetype.json: Each property "description" is present.
INFO: lint-raml-cop: servicepoints.json: Each property "description" is present.
INFO: lint-raml-cop: callnumbertypes.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetypes.json: Each property "description" is present.
INFO: lint-raml-cop: locinsts.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametypes.json: Each property "description" is present.
INFO: lint-raml-cop: modeofissuance.json: Each property "description" is present.
INFO: lint-raml-cop: items.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametype.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcode.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletype.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodes.json: Each property "description" is present.
INFO: lint-raml-cop: classificationtypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemdamagedstatus.json: Each property "description" is present.
ERROR: lint-raml-cop: shelflocation.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocation.json: Missing "description" for: name
INFO: lint-raml-cop: callnumbertype.json: Each property "description" is present.
INFO: lint-raml-cop: loantype.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetypes.json: Each property "description" is present.
ERROR: lint-raml-cop: holdingsrecord.json: Missing "description" for: acquisitionFormat, acquisitionMethod, callNumber, callNumberPrefix, callNumberSuffix, chronology, copyNumber, digitizationPolicy, electronicAccess, entries, enumeration, holdingsInstance, holdingsItems, holdingsStatements, holdingsStatementsForIndexes, holdingsStatementsForSupplements, instanceId, numberOfItems, permanentLocationId, publicDisplay, receiptStatus, receivingHistory, retentionPolicy, shelvingTitle, temporaryLocationId
INFO: lint-raml-cop: loccamp.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstype.json: Each property "description" is present.
INFO: lint-raml-cop: marc.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemdamagedstatuses.json: Each property "description" is present.
ERROR: lint-raml-cop: locations.json: Missing top-level "description".
ERROR: lint-raml-cop: locations.json: Missing "description" for: locations
INFO: lint-raml-cop: illpolicies.json: Each property "description" is present.
INFO: lint-raml-cop: illpolicy.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletypes.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetype.json: Each property "description" is present.
ERROR: lint-raml-cop: shelflocations.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocations.json: Missing "description" for: shelflocations
INFO: lint-raml-cop: locinst.json: Each property "description" is present.
INFO: lint-raml-cop: instances-batch-response.json: Each property "description" is present.
INFO: lint-raml-cop: contributortype.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptype.json: Each property "description" is present.
ERROR: lint-raml-cop: servicepointsuser.json: Missing "description" for: defaultServicePointId, servicePointsIds, userId
INFO: lint-raml-cop: issuancemodes.json: Each property "description" is present.
ERROR: lint-raml-cop: location.json: Missing top-level "description".
ERROR: lint-raml-cop: location.json: Missing "description" for: campusId, code, description, details, discoveryDisplayName, institutionId, isActive, libraryId, name, primaryServicePoint, servicePointIds
INFO: lint-raml-cop: statisticalcodetypes.json: Each property "description" is present.
INFO: lint-raml-cop: time-period.json: Each property "description" is present.
INFO: lint-raml-cop: loantypes.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodetype.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationships.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

bohdan-suprun commented:
> It looks fine to me. I'm not well versed with this relationship domain so I'm not sure if the superInstance and subInstance sample data Ids are in the correct place (they may as well be). What determines which title should be super- or sub-instance?

Hi @mtraneis, preceding title is a child (sub) instance and succeeding is a parent (super) instance. See this comment in the ticket https://issues.folio.org/browse/MODINVSTOR-343?focusedCommentId=59016&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-59016

marcjohnson-kint commented on src/test/java/org/folio/rest/api/ReferenceTablesTest.java:
```diff
@@ -703,6 +706,28 @@ public void statisticalCodesAndCodeTypesBasicCrud()
+  }
+
+  @Test
+  public void instanceRelationshipTypesCrud() throws Exception {
```
What does this test check?

bohdan-suprun replied:
@marcjohnson-kint, it is basic CRUD operations for `/instance-relationship-types`  endpoint, for some reason there was no test present for this endpoint.

marcjohnson-kint replied:
Is this how the other tests for reference records are structured?

bohdan-suprun replied:
@marc-johnson yes, commonly we have one test to check that reference data were loaded to DB and one test to check CRUD operations.

marcjohnson-kint replied:
Ok, if these are the established patterns for this area, then any feedback I might have is likely not relevant for this pull request.

(I think that records created during tests should be deleted prior to the next test, in order to improve the reliability and specificity of tests)

marcjohnson-kint commented on src/test/java/org/folio/rest/api/ReferenceTablesTest.java:
```diff
@@ -703,6 +706,28 @@ public void statisticalCodesAndCodeTypesBasicCrud()
+  public void instanceRelationshipTypesLoaded() throws Exception {
+    Response searchResponse = getReferenceRecords(instanceRelationshipTypesUrl(""));
+    validateNumberOfReferenceRecords("instance-relationship types",
+      searchResponse, 4, 200);
```
If we have a test which checks for the expected count of relationship types, why does this check for a range?

Is there a reason why we aren't sure how many should be loaded?

bohdan-suprun replied:
@marcjohnson-kint, It was a question for me as well. We use such checks for all the reference data tests. 

My guess, that maven test runner do not run tests in the same order every time and because of this there might be test cases that creates additional elements, so we just need some delta to not depend on the order.

marcjohnson-kint replied:
Ok, if these are the established patterns for this area, then any feedback I might have is likely not relevant for this pull request.

(I think that records created during tests should be deleted prior to the next test, in order to improve the reliability and specificity of tests)

mreno-EBSCO replied:
It is best practice that each test should be self contained and execution order should not matter. If order needs to matter, of course, there are ways to do it. But testing ranges instead of exact results yields a fragile test, IMHO. Tests that create things should also clean up those things. Are we not doing there here?

marcjohnson-kint replied:
@mreno-EBSCO I agree and I'd prefer it that way. 

There are quite a few places where that is not done, quite often with regards to the built in sample or reference records.

This pull request doesn't seem to me to be the place to resolve those more broader issues.

bohdan-suprun commented:
Marked with WIP, because we have to merge this PR together with https://github.com/folio-org/mod-inventory/pull/146 to not introduce breaking change.

marcjohnson-kint commented:
> Marked with WIP, because we have to merge this PR together with [folio-org/mod-inventory#146](https://github.com/folio-org/mod-inventory/pull/146) to not introduce breaking change.

I don't understand. How are these changes breaking?

bohdan-suprun commented:
> > Marked with WIP, because we have to merge this PR together with [folio-org/mod-inventory#146](https://github.com/folio-org/mod-inventory/pull/146) to not introduce breaking change.
> 
> I don't understand. How are these changes breaking?

This PR is the 1st part of the preceding-succeeding title feature, here I just added new relationship type.

But in the 2nd part of the feature (https://github.com/folio-org/mod-inventory/pull/146) I've added new preceding/succeedingTitles properties to instance (similar to child/parentInstances), so these properties will be used only for this particular relationship type (preceedingTitles == subInstances, succeedingTitles == superInstances).

And it is possible, that some API consumers/users can create instance relationship with such type and expect to see them in child/parentInstances array. But, when we merge the 2nd part, those relationships will be present in preceding/succeedingTitles array instead of parent/childInstances.

It seems like breaking change for me, is it?

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: mod-inventory-storage/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 72 JSON schema files under directory './ramls'
INFO: lint-raml-cop: loclibs.json: Each property "description" is present.
INFO: lint-raml-cop: loccamps.json: Each property "description" is present.
ERROR: lint-raml-cop: servicepointsuser.json: Missing "description" for: defaultServicePointId, servicePointsIds, userId
INFO: lint-raml-cop: itemdamagedstatuses.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationship.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametype.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationships.json: Each property "description" is present.
INFO: lint-raml-cop: loccamp.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetype.json: Each property "description" is present.
INFO: lint-raml-cop: materialtype.json: Each property "description" is present.
INFO: lint-raml-cop: contributortype.json: Each property "description" is present.
ERROR: lint-raml-cop: shelflocation.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocation.json: Missing "description" for: name
INFO: lint-raml-cop: loantypes.json: Each property "description" is present.
ERROR: lint-raml-cop: uuid.json: Missing top-level "description".
INFO: lint-raml-cop: illpolicies.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatus.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterms.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: servicepoint.json: Each property "description" is present.
INFO: lint-raml-cop: loclib.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatuses.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterm.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetypes.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletypes.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertype.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetypes.json: Each property "description" is present.
INFO: lint-raml-cop: instanceformats.json: Each property "description" is present.
INFO: lint-raml-cop: instancetype.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertypes.json: Each property "description" is present.
INFO: lint-raml-cop: servicepoints.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetype.json: Each property "description" is present.
ERROR: lint-raml-cop: locations.json: Missing top-level "description".
ERROR: lint-raml-cop: locations.json: Missing "description" for: locations
INFO: lint-raml-cop: servicepointsusers.json: Each property "description" is present.
INFO: lint-raml-cop: time-period.json: Each property "description" is present.
INFO: lint-raml-cop: locinsts.json: Each property "description" is present.
INFO: lint-raml-cop: illpolicy.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstype.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetype.json: Each property "description" is present.
INFO: lint-raml-cop: instanceformat.json: Each property "description" is present.
INFO: lint-raml-cop: items.json: Each property "description" is present.
INFO: lint-raml-cop: issuancemodes.json: Each property "description" is present.
INFO: lint-raml-cop: classificationtype.json: Each property "description" is present.
INFO: lint-raml-cop: loantype.json: Each property "description" is present.
INFO: lint-raml-cop: instancetypes.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametypes.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodetypes.json: Each property "description" is present.
ERROR: lint-raml-cop: location.json: Missing top-level "description".
ERROR: lint-raml-cop: location.json: Missing "description" for: campusId, code, description, details, discoveryDisplayName, institutionId, isActive, libraryId, name, primaryServicePoint, servicePointIds
ERROR: lint-raml-cop: holdingsrecord.json: Missing "description" for: acquisitionFormat, acquisitionMethod, callNumber, callNumberPrefix, callNumberSuffix, chronology, copyNumber, digitizationPolicy, electronicAccess, entries, enumeration, holdingsInstance, holdingsItems, holdingsStatements, holdingsStatementsForIndexes, holdingsStatementsForSupplements, instanceId, numberOfItems, permanentLocationId, publicDisplay, receiptStatus, receivingHistory, retentionPolicy, shelvingTitle, temporaryLocationId
INFO: lint-raml-cop: classificationtypes.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcode.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsrecords.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetypes.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletype.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodes.json: Each property "description" is present.
INFO: lint-raml-cop: callnumbertypes.json: Each property "description" is present.
INFO: lint-raml-cop: contributortypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemdamagedstatus.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodetype.json: Each property "description" is present.
ERROR: lint-raml-cop: item.json: Missing "description" for: barcode, chronology, copyNumbers, descriptionOfPieces, electronicAccess, enumeration, holdingsRecord2, holdingsRecordId, itemDamagedStatusDate, itemDamagedStatusId, itemIdentifier, itemLevelCallNumber, itemLevelCallNumberPrefix, itemLevelCallNumberSuffix, itemLevelCallNumberTypeId, materialTypeId, missingPieces, missingPiecesDate, numberOfMissingPieces, numberOfPieces, permanentLoanTypeId, permanentLocationId, staffOnly, temporaryLoanTypeId, temporaryLocationId, volume, yearCaption
ERROR: lint-raml-cop: shelflocations.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocations.json: Missing "description" for: shelflocations
INFO: lint-raml-cop: modeofissuance.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationship.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstypes.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationships.json: Each property "description" is present.
INFO: lint-raml-cop: marc.json: Each property "description" is present.
INFO: lint-raml-cop: materialtypes.json: Each property "description" is present.
INFO: lint-raml-cop: locinst.json: Each property "description" is present.
INFO: lint-raml-cop: instances.json: Each property "description" is present.
INFO: lint-raml-cop: callnumbertype.json: Each property "description" is present.
INFO: lint-raml-cop: instances-batch-response.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptype.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptypes.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>

marcjohnson-kint commented:
> > > Marked with WIP, because we have to merge this PR together with [folio-org/mod-inventory#146](https://github.com/folio-org/mod-inventory/pull/146) to not introduce breaking change.
> > 
> > 
> > I don't understand. How are these changes breaking?
> 
> This PR is the 1st part of the preceding-succeeding title feature, here I just added new relationship type.
> 
> But in the 2nd part of the feature ([folio-org/mod-inventory#146](https://github.com/folio-org/mod-inventory/pull/146)) I've added new preceding/succeedingTitles properties to instance (similar to child/parentInstances), so these properties will be used only for this particular relationship type (preceedingTitles == subInstances, succeedingTitles == superInstances).
> 
> And it is possible, that some API consumers/users can create instance relationship with such type and expect to see them in child/parentInstances array. But, when we merge the 2nd part, those relationships will be present in preceding/succeedingTitles array instead of parent/childInstances.
> 
> It seems like breaking change for me, is it?

I don't think I follow. How can adding a relationship type be a breaking change?

I can understand how separating these out into different properties could be considered a breaking behavioural change. Are you suggesting that it wouldn't be, if it is added at the same time as the relationship type?

id-jenkins commented:
<html><body><pre>
INFO: lint-raml-cop: Investigating and determining configuration: mod-inventory-storage/ramls
INFO: lint-raml-cop: Assessing schema files (https://dev.folio.org/guides/describe-schema/):
INFO: lint-raml-cop: Found 72 JSON schema files under directory './ramls'
INFO: lint-raml-cop: servicepoint.json: Each property "description" is present.
ERROR: lint-raml-cop: uuid.json: Missing top-level "description".
INFO: lint-raml-cop: instancerelationships.json: Each property "description" is present.
INFO: lint-raml-cop: materialtype.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationship.json: Each property "description" is present.
INFO: lint-raml-cop: instanceformat.json: Each property "description" is present.
INFO: lint-raml-cop: materialtypes.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterms.json: Each property "description" is present.
INFO: lint-raml-cop: classificationtype.json: Each property "description" is present.
INFO: lint-raml-cop: instancetype.json: Each property "description" is present.
INFO: lint-raml-cop: instance.json: Each property "description" is present.
INFO: lint-raml-cop: loclib.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsrecords.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetype.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatus.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertypes.json: Each property "description" is present.
INFO: lint-raml-cop: contributortypes.json: Each property "description" is present.
INFO: lint-raml-cop: instancenotetypes.json: Each property "description" is present.
ERROR: lint-raml-cop: item.json: Missing "description" for: barcode, chronology, copyNumbers, descriptionOfPieces, electronicAccess, enumeration, holdingsRecord2, holdingsRecordId, itemDamagedStatusDate, itemDamagedStatusId, itemIdentifier, itemLevelCallNumber, itemLevelCallNumberPrefix, itemLevelCallNumberSuffix, itemLevelCallNumberTypeId, materialTypeId, missingPieces, missingPiecesDate, numberOfMissingPieces, numberOfPieces, permanentLoanTypeId, permanentLocationId, staffOnly, temporaryLoanTypeId, temporaryLocationId, volume, yearCaption
INFO: lint-raml-cop: instanceformats.json: Each property "description" is present.
INFO: lint-raml-cop: identifiertype.json: Each property "description" is present.
INFO: lint-raml-cop: instancetypes.json: Each property "description" is present.
INFO: lint-raml-cop: instancestatuses.json: Each property "description" is present.
INFO: lint-raml-cop: servicepointsusers.json: Each property "description" is present.
INFO: lint-raml-cop: instances.json: Each property "description" is present.
INFO: lint-raml-cop: loclibs.json: Each property "description" is present.
INFO: lint-raml-cop: loccamps.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationship.json: Each property "description" is present.
INFO: lint-raml-cop: natureofcontentterm.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetype.json: Each property "description" is present.
INFO: lint-raml-cop: servicepoints.json: Each property "description" is present.
INFO: lint-raml-cop: callnumbertypes.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetypes.json: Each property "description" is present.
INFO: lint-raml-cop: locinsts.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametypes.json: Each property "description" is present.
INFO: lint-raml-cop: modeofissuance.json: Each property "description" is present.
INFO: lint-raml-cop: items.json: Each property "description" is present.
INFO: lint-raml-cop: contributornametype.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcode.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletype.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodes.json: Each property "description" is present.
INFO: lint-raml-cop: classificationtypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemdamagedstatus.json: Each property "description" is present.
ERROR: lint-raml-cop: shelflocation.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocation.json: Missing "description" for: name
INFO: lint-raml-cop: callnumbertype.json: Each property "description" is present.
INFO: lint-raml-cop: loantype.json: Each property "description" is present.
INFO: lint-raml-cop: itemnotetypes.json: Each property "description" is present.
ERROR: lint-raml-cop: holdingsrecord.json: Missing "description" for: acquisitionFormat, acquisitionMethod, callNumber, callNumberPrefix, callNumberSuffix, chronology, copyNumber, digitizationPolicy, electronicAccess, entries, enumeration, holdingsInstance, holdingsItems, holdingsStatements, holdingsStatementsForIndexes, holdingsStatementsForSupplements, instanceId, numberOfItems, permanentLocationId, publicDisplay, receiptStatus, receivingHistory, retentionPolicy, shelvingTitle, temporaryLocationId
INFO: lint-raml-cop: loccamp.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstype.json: Each property "description" is present.
INFO: lint-raml-cop: marc.json: Each property "description" is present.
INFO: lint-raml-cop: holdingstypes.json: Each property "description" is present.
INFO: lint-raml-cop: itemdamagedstatuses.json: Each property "description" is present.
ERROR: lint-raml-cop: locations.json: Missing top-level "description".
ERROR: lint-raml-cop: locations.json: Missing "description" for: locations
INFO: lint-raml-cop: illpolicies.json: Each property "description" is present.
INFO: lint-raml-cop: illpolicy.json: Each property "description" is present.
INFO: lint-raml-cop: alternativetitletypes.json: Each property "description" is present.
INFO: lint-raml-cop: holdingsnotetype.json: Each property "description" is present.
ERROR: lint-raml-cop: shelflocations.json: Missing top-level "description".
ERROR: lint-raml-cop: shelflocations.json: Missing "description" for: shelflocations
INFO: lint-raml-cop: locinst.json: Each property "description" is present.
INFO: lint-raml-cop: instances-batch-response.json: Each property "description" is present.
INFO: lint-raml-cop: contributortype.json: Each property "description" is present.
INFO: lint-raml-cop: instancerelationshiptype.json: Each property "description" is present.
ERROR: lint-raml-cop: servicepointsuser.json: Missing "description" for: defaultServicePointId, servicePointsIds, userId
INFO: lint-raml-cop: issuancemodes.json: Each property "description" is present.
ERROR: lint-raml-cop: location.json: Missing top-level "description".
ERROR: lint-raml-cop: location.json: Missing "description" for: campusId, code, description, details, discoveryDisplayName, institutionId, isActive, libraryId, name, primaryServicePoint, servicePointIds
INFO: lint-raml-cop: statisticalcodetypes.json: Each property "description" is present.
INFO: lint-raml-cop: time-period.json: Each property "description" is present.
INFO: lint-raml-cop: loantypes.json: Each property "description" is present.
INFO: lint-raml-cop: statisticalcodetype.json: Each property "description" is present.
INFO: lint-raml-cop: electronicaccessrelationships.json: Each property "description" is present.
INFO: lint-raml-cop: Not assessing RAML/Schema or examples against schema, as per option '--json-only'.
ERROR: lint-raml-cop: There were processing errors. See list above.
</pre><body></html>
